qiskit_experiments/data_processing/nodes.py

```diff
@@ -56,47 +58,216 @@ def _format_data(self, datum: Any) -> Any:
                 or averaged IQ date (two-dimensional).
 
         Returns:
-            datum as a numpy array.
+            datum and any error estimate as a numpy array.
 
         Raises:
             DataProcessorError: If the datum does not have the correct format.
         """
         datum = np.asarray(datum, dtype=float)
 
+        if error is not None:
+            error = np.asarray(error, dtype=float)
+
         if self._validate and len(datum.shape) != self._required_dimension():
             raise DataProcessorError(
-                f"Single-shot data given {self.__class__.__name__}"
+                f"Single-shot data given to {self.__class__.__name__}"
                 f"must be a {self._required_dimension()}D array. Instead, a {len(datum.shape)}D "
                 f"array was given."
             )
 
-        return datum
+        if error is not None and self._validate and len(error.shape) != self._required_dimension():
+            raise DataProcessorError(
+                f"Erorr on single-shot data given to {self.__class__.__name__}"
+                f"must be a {self._required_dimension()}D array. Instead, a {len(error.shape)}D "
+                f"array was given."
+            )
+
+        return datum, error
 
     def __repr__(self):
         """String representation of the node."""
         return f"{self.__class__.__name__}(validate: {self._validate}, scale: {self.scale})"
 
 
+class AverageIQData(IQPart):
```

Comment:
There is nothing special about IQ data for this, no?

Reply:
Indeed, I removed any reference to IQ data and made it a bit more general. See here: 35f903f